File: scratch/image_dims.js
const fs = require('fs');
const path = require('path');

function getPngSize(buffer) {
    const width = buffer.readUInt32BE(16);
    const height = buffer.readUInt32BE(20);
    return { width, height };
}

function getJpgSize(buffer) {
    let i = 2; // skip SOI
    while (i < buffer.length) {
        // Find next marker
        while (i < buffer.length && buffer[i] !== 0xFF) {
            i++;
        }
        while (i < buffer.length && buffer[i] === 0xFF) {
            i++;
        }
        if (i >= buffer.length) break;
        const marker = buffer[i];
        i++;
        if (marker === 0xD9 || marker === 0xDA) { // EOI or SOS
            break;
        }
        const size = buffer.readUInt16BE(i);
        if (marker >= 0xC0 && marker <= 0xC3) { // SOF0 - SOF3
            const height = buffer.readUInt16BE(i + 3);
            const width = buffer.readUInt16BE(i + 5);
            return { width, height };
        }
        i += size;
    }
    return null;
}

const dir = 'e:\\Websites\\Skyom\\assets\\Skyom Floor';
if (fs.existsSync(dir)) {
    const files = fs.readdirSync(dir);
    files.forEach(file => {
        const filepath = path.join(dir, file);
        if (fs.statSync(filepath).isFile()) {
            const buf = fs.readFileSync(filepath);
            let size = null;
            if (buf[0] === 0x89 && buf[1] === 0x50 && buf[2] === 0x4E && buf[3] === 0x47) {
                size = getPngSize(buf);
            } else if (buf[0] === 0xFF && buf[1] === 0xD8) {
                size = getJpgSize(buf);
            }
            if (size) {
                console.log(`${file}: ${size.width}x${size.height} (${(size.width/size.height).toFixed(2)})`);
            } else {
                console.log(`${file}: Unknown format`);
            }
        }
    });
} else {
    console.log("Directory not found");
}

File: scratch/image_dims.py
import os
import struct

def get_image_info(filepath):
    """Determine the image type and dimensions."""
    size = os.path.getsize(filepath)
    with open(filepath, 'rb') as f:
        data = f.read(30)
        
    if data.startswith(b'\x89PNG\r\n\x1a\n'):
        # PNG
        w, h = struct.unpack('>LL', data[16:24])
        return 'PNG', int(w), int(h)
    elif data.startswith(b'\xff\xd8'):
        # JPEG
        with open(filepath, 'rb') as f:
            f.seek(0)
            f.read(2)
            b = f.read(1)
            try:
                while (b and b != b'\xda'):
                    while (b != b'\xff'): b = f.read(1)
                    while (b == b'\xff'): b = f.read(1)
                    if (b >= b'\xc0' and b <= b'\xc3'):
                        f.read(3)
                        h, w = struct.unpack('>HH', f.read(4))
                        return 'JPEG', int(w), int(h)
                    else:
                        chunk_len = struct.unpack('>H', f.read(2))[0]
                        f.read(chunk_len - 2)
                    b = f.read(1)
            except Exception as e:
                return 'JPEG', 0, 0
    return 'Unknown', 0, 0

folder = r'e:\Websites\Skyom\assets\Skyom Floor'
if os.path.exists(folder):
    print("Files in Skyom Floor:")
    for filename in sorted(os.listdir(folder)):
        filepath = os.path.join(folder, filename)
        if os.path.isfile(filepath):
            t, w, h = get_image_info(filepath)
            print(f"{filename}: {t} {w}x{h}")
else:
    print("Folder does not exist")
